src/components/flow/routers/giftcard/GiftCardRouterForm.tsx

```diff
@@ -40,15 +40,29 @@ class GiftCardRouterForm extends React.PureComponent<RouterFormProps, GiftCardRo
     super(props);
     this.state = nodeToState(this.props.nodeSettings);
   }
-  private handleSave = () => {
-    if (this.state.valid) {
+
+  private handleSave = (): void => {
+    // validate in case they never updated an empty field
+    const updates: Partial<GiftCardRouterFormState> = {
+      giftcardDb: validate('Giftcard Database', this.state.giftcardDb.value, [Required]),
+      resultName: validate('Result Name', this.state.resultName.value, [
+        Alphanumeric,
+        StartIsNonNumeric,
+        Required
+      ])
+    };
+
+    const updated = mergeForm(this.state, updates);
+    this.setState(updated);
+
+    if (this.state.valid && updated.valid) {
       this.props.updateRouter(stateToNode(this.props.nodeSettings, this.state));
       this.props.onClose(false);
     }
   };
 
   private handleUpdateAssignChange = (value: SelectOption, submitting = false): void => {
-    const giftcardType = validate('giftcardType', value.value, [shouldRequireIf(submitting)]);
+    const giftcardType = validate('Giftcard Database', value.value, [shouldRequireIf(submitting)]);
```

Comment:
`giftcardType` is a different field from `giftcardDb`, I think here you can keep `giftcardType` since this field does not have search functionality, it's a normal select field.

Reply:
I've changed it to "GIftcard Type"